nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/functions/PropertyUtils.java

```diff
@@ -126,7 +126,7 @@ public static <T> T getPropertyValue(DocumentModel doc, String propertyName, Cla
      */
     public static String base64EncodeBlob(Blob blob) {
         try {
-            if (blob != null) {
+            if (blob != null && blob.getByteArray() != null && blob.getByteArray().length > 0) {
                 return Base64.encodeBase64String(blob.getByteArray());
             }
```

Comment:
An `InputStream` is created/fetched for each call to `getByArray`, this may be not performant to call it multiple times (depending on the blob implementation).
You should store the result:
```suggestion
            if (blob != null) {
                byte[] blobBytes = blob.getByteArray();
                 if (ArrayUtils.isNotEmpty(blobBytes)) {
                    return Base64.encodeBase64String(blobBytes);
                }
             }
```